### PR TITLE
passwordstore: Add some real gopass integration tests

### PR DESCRIPTION
**Pull request by grembo**
##### SUMMARY
Since the lookup_passwordstore module gained some real gopass support, add real unit tests for it.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lookup_passwordstore

### CODE REVIEW

**Comment by grembo** [comment]
@felixfontein This is ready to be reviewed now.

File: changelogs/fragments/5030-add-gopass-tests-to-lookup-passwordstore-plugin.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - passwordstore lookup plugin - add integration tests using (real) gopass (https://github.com/ansible-collections/community.general/pull/5130).
```
Comment (felixfontein) [inline changelogs/fragments/5030-add-gopass-tests-to-lookup-passwordstore-plugin.yml:2]: Please remove the changelog fragment. Test-only changes should not have one :)

File: tests/integration/targets/lookup_passwordstore/tasks/package.yml [hunk 2]
```diff
@@ -31,6 +31,29 @@
       disable_gpg_check: yes
   when: ansible_facts.pkg_mgr in ['zypper', 'community.general.zypper']
 
+# See https://github.com/gopasspw/gopass/issues/1849#issuecomment-802789285
+- name: Install on Debian
+  when: ansible_facts.os_family == 'Debian'
+  become: yes
+  block:
+  - name: Fetch gopass repo keyring
+    shell: curl -o /usr/share/keyrings/gopass-archive-keyring.gpg \
```
Comment (felixfontein) [inline tests/integration/targets/lookup_passwordstore/tasks/package.yml:40]:
```suggestion
    command: curl -o /usr/share/keyrings/gopass-archive-keyring.gpg \
```

File: tests/integration/targets/lookup_passwordstore/tasks/package.yml [hunk 3]
```diff
@@ -31,6 +31,29 @@
...
+        Types: deb
+        URIs: https://packages.gopass.pw/repos/gopass
+        Suites: stable
+        Architectures: amd64 arm64 armhf
+        Components: main
+        Signed-By: /usr/share/keyrings/gopass-archive-keyring.gpg
+  - name: Update repo
+    command: apt update
+  - name: Install package from gopass repo
+    command: apt install -y gopass
```
Comment (felixfontein) [inline tests/integration/targets/lookup_passwordstore/tasks/package.yml:55]: Why not use the [apt](https://docs.ansible.com/ansible/devel/collections/ansible/builtin/apt_module.html#ansible-collections-ansible-builtin-apt-module) and [apt_repository](https://docs.ansible.com/ansible/devel/collections/ansible/builtin/apt_repository_module.html#ansible-collections-ansible-builtin-apt-repository-module) modules for this?

File: tests/integration/targets/lookup_passwordstore/tasks/tests.yml [hunk 4]
```diff
@@ -70,150 +80,22 @@
...
-    that:
-      - readpass == newpass
-
-- name: Try to read folder as passname
-  set_fact:
-    newpass: "{{ lookup('community.general.passwordstore', 'folder') }}"
-  ignore_errors: true
-  register: eval_error
+- name: Initialise gopass passwordstore
+  command: gopass init --path {{ lookup('env','HOME') }}/.gopass-store ansible-test
```
Comment (felixfontein) [inline tests/integration/targets/lookup_passwordstore/tasks/tests.yml:87]: This should probably work:
```suggestion
  command: gopass init --path $HOME/.gopass-store ansible-test
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
I'll merge this later today if nobody complains :)

**Comment by felixfontein** [comment]
@grembo thanks for contributing these tests!

**Comment by grembo** [comment]
> @grembo thanks for contributing these tests!

@felixfontein Thanks for reviewing them!

**Pull request merged**